monitoring/monitorlib/messagesigning/message_signer.py

```diff
@@ -0,0 +1,104 @@
+import base64
+import time
+from Crypto.PublicKey import RSA
+from Crypto.Signature import pkcs1_15
+from Crypto.Hash import SHA256
+from loguru import logger
+import urllib.parse
+from monitoring.monitorlib.messagesigning.hasher import get_content_digest
+from monitoring.monitorlib.messagesigning.uft import get_x_utm_jws_header
+
+def get_signed_headers(object_to_sign, private_key_path, cert_url):
+    signed_type = str(type(object_to_sign))
+    is_signing_request = "PreparedRequest" in signed_type
+    sig, sig_input = _get_signature(object_to_sign, signed_type, private_key_path)
+    content_digest = (
+        get_content_digest(object_to_sign.body)
+        if is_signing_request
+        else get_content_digest(object_to_sign.get_data())
+    )
+    signed_headers = {
+        "x-utm-message-signature-input": "utm-message-signature={}".format(sig_input),
+        "x-utm-message-signature": "utm-message-signature=:{}:".format(sig),
+        "x-utm-jws-header": get_x_utm_jws_header(cert_url),
```

Comment:
It's not wrong but seems a little weird to compute the x-utm-jws-header twice for each signature -- redundant calculations often introduce the possibility of inconsistency.  Why not compute the header once and pass it to the functions that need it (`_get_signature` here because it needs to provide it to `_get_signature_base`), as that would have the added benefit of making the low-level signature computation more agnostic of specific conventions chosen by UFT?